test/tsconfig.json

```diff
@@ -1,4 +1,11 @@
 {
-  "extends": "../tsconfig"
+  	"extends": "../tsconfig",
+	"compilerOptions": {
+		"rootDir": ".."
+	},
+	"include": [
+		"../src/**/*",
+		"../test/**/*"
+	]
 }
```

Comment:
After a few attempts... I finally fixed things so that IntelliJ doesn't complain about the test utilities not being under the root directory. Note that we can have a tsconfig.json file inside of the test folder that has acts independently of the one under the root directory. So for regular build purposes, we do not include the test folder. But for test purposes we do.